1-raw-data/mailinglists/googlegroups/googlegroups_data_test.go

```diff
@@ -0,0 +1,297 @@
+// limitations under the License.
+
+package googlegroups
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/PuerkitoBio/goquery"
+)
+
+func TestGetHttpStringResponse(t *testing.T) {
+	var gotResponseString string
+	var gotErr error
+
+	tests := []struct {
+		comparisonType     string
+		url                string
+		wantResponseString string
+		wantErr            error
+	}{
+		{
+			comparisonType:     "Get empty response if no url",
+			url:                "",
+			wantResponseString: "",
+			wantErr:            nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.comparisonType, func(t *testing.T) {
+
+			if gotResponseString, gotErr = httpStringResponse(test.url); !errors.Is(gotErr, test.wantErr) {
+				t.Errorf("Error response does not match.\n got: %v\nwant: %v", gotErr, test.wantErr)
+			}
+			if strings.Compare(gotResponseString, test.wantResponseString) != 0 {
+				t.Errorf("Filename response does not match.\n got: %v\nwant: %v", gotResponseString, test.wantResponseString)
+			}
+		})
+	}
+}
+
+// TODO test for error?
+func TestGetFileName(t *testing.T) {
+	var gotDateKey string
+	var gotErr error
+
+	tests := []struct {
+		comparisonType string
+		matchDate      string
+		wantFileName   string
+		wantErr        error
+	}{
+		{
+			// Confirm correct output with 1 dig month and 1 dig day
+			comparisonType: "Single digit month and single digit day",
+			matchDate:      "1/29/91",
+			wantFileName:   "1991-01.txt",
+			wantErr:        nil,
+		},
+		{
+			// Confirm correct output with 2 dig month and 1 dig day and not future
+			comparisonType: "Double digit month and Single digit day",
+			matchDate:      "9/2/38",
+			wantFileName:   "1938-09.txt",
+			wantErr:        nil,
+		},
+		{
+			// Confirm correct output with 1 dig month and 2 dig day
+			comparisonType: "Single digit month and double digit day",
+			matchDate:      "1/24/95",
+			wantFileName:   "1995-01.txt",
+			wantErr:        nil,
+		},
+		{
+			// Confirm correct output with 2 dig month and 2 dig day
+			comparisonType: "Double digit month and double digit day",
+			matchDate:      "11/11/17",
+			wantFileName:   "2017-11.txt",
+			wantErr:        nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.comparisonType, func(t *testing.T) {
+			if gotDateKey, gotErr = getFileName(test.matchDate); !errors.Is(gotErr, test.wantErr) {
+				t.Errorf("Error response does not match.\n got: %v\nwant: %v", gotErr, test.wantErr)
+			}
+			if strings.Compare(gotDateKey, test.wantFileName) != 0 {
+				t.Errorf("Filename response does not match.\n got: %v\nwant: %v", gotDateKey, test.wantFileName)
+			}
+		})
+	}
+}
+
+func TestGetTotalTopics(t *testing.T) {
+	exCorrectResponseBody := `
+	<html>
+	<head>
+		<title>My document</title>
+	</head>
+		<i>Showing 1-100 of 1891 topics</i>
+  </html>
+		`
+	exMissingResponseBody := `
+	<html>
+	<head>
+		<title>My document</title>
+	</head>
+  </html>
+		`
```

Comment:
Has this been gofmt'ed?

Reply:
Yes I did run gofmt on the files but it looks like that doesn't work on this segment.